src/lib.rs

```diff
@@ -59,7 +59,11 @@
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
+#[macro_use]
+extern crate alloc;
```

Comment:
All this should probably be hidden behind an (default-enabled and enabled by the `std` feature) `alloc` feature. Not all non-std platforms have alloc

Reply:
Ah! also `Vec::from_raw_parts` support is added in 1.37.0 which is quite recent, so how should we handle that as wel if someone is using it on older versions?

Reply:
We bump the minimum version to 1.37. That's fine